lib/animas/animasSimulator.js

```diff
@@ -85,9 +90,53 @@ exports.make = function(config){
     events.push(e);
   }
 
+  function checkIfBasalSuspendedByAlarm(alarm,basal) {
+    if ((basal !== null) && (alarm != null) &&
+        (basal.rate === 0) && (basal.deliveryType !== 'suspend')) {
+
+      // Animas does not generate suspend/resume events for alarms,
+      // so we check if an alarm occurred recently or at the same time.
+      // If so, we mark the basal as suspended and generate a new suspend event.
+
+      var duration = Date.parse(basal.time) - Date.parse(alarm.time);
+      console.log(alarm, basal);
+      var twoHours = (120 * sundial.MIN_TO_MSEC);
+      if(duration < twoHours) {
+
+        var type = alarm.alarmType;
+        if(type === 'occlusion' || type === 'auto_off' || type === 'no_insulin' || type === 'no_power') {
+
+          //this basal is suspended
+          basal.deliveryType = 'suspend';
+          annotate.annotateEvent(basal, 'animas/basal/marked-suspended-from-alarm');
+
+          var status = {
+            time: basal.time,
+            deviceTime: basal.deviceTime,
```

Comment:
@jebeck I think the issue is not with `deviceTime`, but with `time`. I wasn't sending the most recent record to `TZOUtil`, which meant that `time` and `deviceTime` would always be the same. I'm now sending the most recent event time (final suspend) to `TZOUtil` which should fix this issue. However, while investigating `deviceTime` I did find one place where it is incorrect: manual smbg events.